chapters/chapter-abstracts.md

```diff
@@ -0,0 +1,147 @@
+## 5. Building Command-Line Tools with Python
+
+Writing command-line Python scripts allows us to run our programs from the Unix shell,
+which facilitates both scalability and interoperability with data analysis pipelines.
+The `argparse` Python module can be used to parse user input,
+and provides other expected command line functionality
+such as positional and optional arguments, and help messages.
+Python scripts can take input from files or directly from the standards input
+and write to fies or the standard output.
+When we write Python scripts we put the put `import` statements first,
+and use a special variable called `__name__`
+to check whether the script is run as the main program
+or called from another program.
+Instead of having one long script for all code,
+it is recommended to modularize code into different files 
+according to functionality
+and to document the behavior of each module and function carefully
+in their respective docstring.
+
+## 9. Automating Analyses with Make
+
+Keeping track of programs that process multiple data files
+is an error prone process that we should automate to minimize mistakes.
+Build-managers such as `make`
+can keep track of file dependencies in the analysis pipelines
+and run our programs automatically to update files that are out of date.
+A build rule has targets, prerequisites, and a recipe.
+A target can be a file or a phony target that triggers an action.
+When a target is out of date with respect to its prerequisites,
+Make executes the recipe associated with its rule.
+Make defines a few convenience variables
+such as `$@` (target),
+`$^` (all prerequisites),
+and `$<` (first prerequisite).
+Makefiles can also use wildcard matching of file names
+and functions such as `$(patsubst ...)` for increased flexibility.
+As always it is important to document our code
+and in Make we use specially-formatted comments to create self-documenting Makefiles.
+
+## 10. Configuring Programs
+
+For complex programs
+it can improve usability and increase reproducibility
+to be able to read configuration options from files
+in addition to using command-line options.
```

Comment:
Perhaps this sentence would read better as the following?
"For complex programs, usability and reproducibility can be improved by reading configuration options from files in addition to using command-line options."